sourcecode/www/src/components/ResourceModal/ResourceStats.jsx

```diff
@@ -8,71 +8,100 @@ import {
     XAxis,
     YAxis,
 } from 'recharts';
-import { fillEmptyDays } from '../../helpers/chart.js';
+import {fillEmptyDays} from '../../helpers/chart.js';
 import useFetchWithToken from '../../hooks/useFetchWithToken.jsx';
-import { Box, CircularProgress } from '@mui/material';
+import {Box, CircularProgress} from '@mui/material';
 import moment from 'moment';
 import useTranslation from '../../hooks/useTranslation.js';
-import { useConfigurationContext } from '../../contexts/Configuration.jsx';
+import {useConfigurationContext} from '../../contexts/Configuration.jsx';
 
-const ResourceStats = ({ resourceId }) => {
-    const { t } = useTranslation();
-    const { edlibApi } = useConfigurationContext();
+const ResourceStats = ({resourceId}) => {
+    const {t} = useTranslation();
+    const {edlibApi} = useConfigurationContext();
 
-    const { loading, response } = useFetchWithToken(
-        edlibApi(`/resources/v1/resources/${resourceId}/stats`)
+    const today = moment(new Date()).format('YYYY-MM-DD');
+    const lastMonth = moment(today).subtract(30, 'days').format('YYYY-MM-DD');
+
+    const [startDate, setStartDate] = useState(lastMonth);
+    const [endDate, setEndDate] = useState(today);
+
+    const {loading, response} = useFetchWithToken(
+        edlibApi(`/resources/v1/resources/${resourceId}/stats?start=${startDate}&end=${endDate}`)
     );
 
-    if (loading || !response) {
-        return <CircularProgress />;
-    }
-    const last7daysViews = response.data.last7daysViews;
+    const dateRangeViews = response?.data?.dateRangeViews || {};
     const datasets = [
         {
             key: 'count',
-            name: _.capitalize(t('resource_view', { count: 2 })),
-            dataset: last7daysViews,
+            name: _.capitalize(t('resource_view', {count: 2})),
+            dataset: dateRangeViews,
         },
     ];
 
+    const handleDateChange = (event) => {
+        setStartDate(event.target.value);
+    };
+    const toDateChange = (event) => {
+        setEndDate(event.target.value);
+    };
+
     return (
         <>
             <Box pb={2}>
-                <strong>{t('S.VIEWS_PAST_7_DAYS')}</strong>
+                <strong>{t('S.VIEWS')}</strong>
+            </Box>
+
+            <Box pb={2}>
+                From: <label htmlFor="from"> <input type="date" id="from" name="from" max={endDate}
```

Comment:
The label text has to be *inside* `<label>`.

The `name` property isn't needed, since this isn't a real HTML form.